### DISCUSSION TITLE
v2 - 2022-02-04

### DISCUSSION THREAD

**Discussion by TheRealAyCe** [TheRealAyCe/ubisoftconnect-win7fix#5] [Announcements]
- Small changes/improvements.
- Requests window

<hr /><em>This discussion was created from the release <a href='https://github.com/TheRealAyCe/ubisoftconnect-win7fix/releases/tag/v2'>v2 - 2022-02-04</a>.</em>

**Comment by Syntaxrabbit** [comment]
Great job! Any chance we could leave out java completely? I am thinking of running a dotnet(core) application on a separate Linux/Windows machine. I can code in VS2022, but my java knowledge is quite limited despite the similar syntax. My last java application is about 15 years old...

**Reply by TheRealAyCe** [reply]
Java is not required to build a tool like this, but I've used it here because it does not rely on the OS's library for making HTTP calls and it can also be prevented from refreshing its hosts<->IP address cache. That way I can host the webserver and make the real request on the same machine. .NET would fail to contact the Ubisoft server just as Ubisoft Connect fails.

Instead of my fix you could build your own tool, maybe using some of my C# code as a base. You can do this:
1. On your Win7 machine add a hosts file entry to redirect the Ubisoft endpoint to your Win10/Linux machine's IP address.
2. Host a HTTPS webserver app on that machine which pretends to be the endpoint, using a self-signed certificate.
3. Install that certificate on the Win7 machine too, so that Ubisoft Connect accepts your server.
4. For each request your HTTPS webserver app receives, simply call the real Ubisoft endpoint with the provided request and pipe back its response. Maybe some headers need to be changed/stripped.

**Comment by EA-LOST** [comment]
thanks for you app it really helps a lot . but the problem is i can invite my friends but i can't open the chat in ubisoft connect overlay.
Chat in the ubisoft connect app works fine , requests in your app shows that message through ubisoft connect app sended correctly and completed . but when i want to chat from Ubisoft Connect overlay it says [Chat is currently unavailable] . i checked requests in your app but there was nothing like failed request or anything ,chat just didn't work. the interesting thing is when i use your app, i can invite my friends but i can't chat through overlay , but when i don't use your app i can chat but i can't invite my friends -_-

**Reply by TheRealAyCe** [reply]
Yep, sadly the Overlay does not seem work with it. If my app doesn't even report a request when you do something through the overlay, it does not sound like something that I can fix. :(

**Reply by EA-LOST** [reply]
![Ubibug](https://user-images.githubusercontent.com/62942032/189769885-c833935e-f183-4815-9e37-7c92907f6193.PNG)


that's unfortunate :( , Ubisoft included Windows7 in the specification page of the game but they're not responsible for their customers. 


in games like R6 you can join to someone's lobby from overlay but in For Honor you can't do that and you have to invite someone so they could join, inviting works with your app and i'm very grateful and thanks for that =D

**Reply by TheRealAyCe** [reply]
Yep, I too think they shouldn't advertise compatibility with Win7 when they stopped supporting it. The problem that my app fixes could be fixed extremely simply by Ubisoft, but they ignored it.

Have fun!

**Comment by electrophreak** [comment]
Great work helping support Win 7!
Anybody seen a fix for this.  I've tried most of the standard ones found out on the web.
Dialog: upc.exe
"The procedure entry point CreateDXGIFactory2 could not be located in the dynamic link library dxgi.dll"

I had this same error in the past, but game always went ahead and start.  Now nothing.  I hit OK on the dialog and nothing happens.
BTW: This is a steam game.
TIA!

**Reply by EA-LOST** [reply]
well i've a workaround for that error
you've to go to ubisoft connect directory (mine is this D:\Ubisoft Connect\Ubisoft Game Launcher)
your's probably is in Drive C ,anyway

after that delete the upc.exe
and then find upc.exe.old
and rename upc.exe.old to upc.exe

and here you are it will work just like before =D

**Reply by electrophreak** [reply]
Thank you!  I couldn't find the "old" file, so I right clicked on the folder and used "Restore previous versions".  Copied the file from the oldest "Previous version" folder and it worked.
